Database

Andebugulin opened the pull request:
### UPDATES:

- Added database schema SQL script according to sqema, there:
  -  creating all `TABLES` according to schema
  - inserting `EXAMPLES` to the tables
- updated `postgres.yml` for automating usage of SQL (creating and inserting) script
- automated `./start-services` script for accessing db in the terminal right away, without using localhost:9999

Andebugulin committed:
feature: auto-creating-sql-schema-inserting-example-data

when starting docker compose it will create data base using sql script

using sql script it will as well insert some example data to the database

TODO: add to start-service script ability to access postgres CLI, so that database can be managed via terminal

Andebugulin committed:
feature: added-manage_db-command-to-the-script-to-automate-database-cli

now using <./start-service manage_db> command you will get access to the database via terminal

martimolanes commented:
> updated postgres.yml for automating usage of SQL (creating and inserting) script

it's an interesting approach, i really like it! Thank you!

> automated ./start-services script for accessing db in the terminal right away, without using localhost:9999

works flawlessly, i really like if the format of the start-service manage_db is like the start-service. Like `./start-service.sh manage-db`

>Added database schema SQL script according to sqema, there:
    creating all TABLES according to schema
    inserting EXAMPLES to the tables

This needs more work to be done, for the moment, this doesn't achieve our specificacion ( you can see it in `docs/database-schema.md` ). The problem is that you are not using the Tags table you specify, for the Task and Activity Table. To change this we need probably to incorporate two new tables where we can store the relation between Tags and Activities/Tasks. **Probably change our specificacion also**. Very good work btw! keep it up :)

martimolanes commented on client/bun.lockb:
dont edit this file in this PR

martimolanes commented on database/init-db.sql:
```diff
@@ -0,0 +1,82 @@
+    StartDate TIMESTAMP,
+    EndDate TIMESTAMP,
+    Status INTEGER REFERENCES Status(Id),
+    Tags TEXT, -- Consider using an association table for a normalized design
```
```suggestion

```
maybe delete this

martimolanes commented on database/init-db.sql:
```diff
@@ -0,0 +1,82 @@
+    Content TEXT,
+    StartDate TIMESTAMP,
+    EndDate TIMESTAMP,
+    Tags TEXT, -- Consider using an association table for a normalized design
```
```suggestion
    Tags TEXT, -- Consider using an association table for a normalized design
```
and this

martimolanes commented on database/init-db.sql:
```diff
@@ -0,0 +1,82 @@
+    Status INTEGER REFERENCES Status(Id),
+    ActivityId INTEGER REFERENCES Activity(Id)
+);
+
```
```suggestion

CREATE TABLE IF NOT EXISTS ActivityTag (
    ActivityId INTEGER REFERENCES Activity(Id),
    TagId INTEGER REFERENCES Tag(Id),
    PRIMARY KEY (ActivityId, TagId)
);

CREATE TABLE IF NOT EXISTS TaskTag (
    TaskId INTEGER REFERENCES Task(Id),
    TagId INTEGER REFERENCES Tag(Id),
    PRIMARY KEY (TaskId, TagId)
);
```
and add here two tables like this ones

Andebugulin replied:
fixed

martimolanes commented on docs/database-usage.md:
```diff
@@ -0,0 +1,52 @@
+
+    -  System: `PostgreSQL`
+    -  Server: `db`
+    - Username: `postgres` (the default superuser account created by the PostgreSQL Docker image)
```
fix format

martimolanes commented on start-service.sh:
```diff
@@ -9,6 +9,7 @@ function help() {
     echo "usage:"
     echo "  run:"
     echo "    $0 db               - start PostgreSQL database"
+    echo "    $0 manage_db        - start a PostgreSQL terminal session"
```
it would be better to have the same format that the script name. `manage-db`

Andebugulin replied:
fixed

martimolanes commented on server/bun.lockb:
dont edit this file also

Andebugulin replied:
even tho, I copy from main branch bun.lockb files and past them in a database branch, and add them with 
git add client/bun.lockb server/bun.lockb  
  1) git doesn't recognize any changes
  2) any suggestions?

Andebugulin committed:
fix: schema-script

schema has been adjusted to use junction tables

schema docs have been updated

in script manage_db was change to manage-db

Andebugulin committed:
updated: database-usage.md

martimolanes reviewed (approved):
LGTM!
